### ISSUE TITLE
TabNetClassifier using softmax activation for both binary and multi-class classification

### ISSUE THREAD

**Issue by pmwaniki** [dreamquark-ai/tabnet#313]
**Describe the bug**

**What is the current behavior?**
The predict_proba function in TabNetClassifier uses softmax activation for both binary and multi-class classification
**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

TabNetClassifier should use sigmoid activation if output_dim=1
**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

**Comment by pmwaniki** [comment]
Sigmoid activation is not needed if output_dim is set to 2 for binary classification. TabNetClassifier should raise an error if output_dim=1

**Comment by Optimox** [comment]
Hello,

Could you explain a bit more what is the problem? Output dim is the number of classes you have in your training data (binary or not). There is no way to train a classifier with only one class (a trivial model), and I think there is already an error if you try to train a classifier with only one class.

Could you share a minimal code to reproduce the error you are talking about?

**Comment by pmwaniki** [comment]
X=np.random.randn(100,5)
y=np.random.choice([0.,1.],size=100,replace=True)

clf=TabNetClassifier(output_dim=1)
clf.fit(X,y)
clf.predict_proba(X)

The TabNetClassifier still allows one to specify output_dim=1. But i guess the internal behaviour is correct.
